nx-dev/nx-dev/pages/packages/index.tsx

```diff
@@ -1,39 +1,52 @@
-import { sortCorePackagesFirst } from '@nrwl/nx-dev/data-access-packages';
-import { MenuItem, MenuSection } from '@nrwl/nx-dev/models-menu';
+import { Heading1 } from '@nrwl/nx-dev-feature-package-schema-viewer';
+import { getPackagesSections } from '@nrwl/nx-dev/data-access-menu';
+import {
+  filterMigrationPackages,
+  sortCorePackagesFirst,
+} from '@nrwl/nx-dev/data-access-packages';
+import { Menu, MenuItem, MenuSection } from '@nrwl/nx-dev/models-menu';
+import { IntrinsicPackageMetadata } from '@nrwl/nx-dev/models-package';
 import {
   Breadcrumbs,
   DocumentationHeader,
   Footer,
   SidebarContainer,
 } from '@nrwl/nx-dev/ui-common';
+import { iconsMap } from '@nrwl/nx-dev/ui-references';
 import { NextSeo } from 'next-seo';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { useMemo } from 'react';
-import { Heading1 } from '../../feature-package-schema-viewer/src/lib/ui/headings';
-import { iconsMap } from '../../ui-references/src/lib/icons-map';
-import { nxMenuApi } from '../lib/api';
-import { useNavToggle } from '../lib/navigation-toggle.effect';
+import { menusApi } from '../../lib/menus.api';
+import { useNavToggle } from '../../lib/navigation-toggle.effect';
+import { nxPackagesApi } from '../../lib/packages.api';
 
-interface ReferencesProps {
-  references: MenuSection;
-}
+export default function Packages({
+  packages,
+  menu,
+}: {
+  packages: IntrinsicPackageMetadata[];
+  menu: MenuItem[];
+}): JSX.Element {
+  const router = useRouter();
+  const { toggleNav, navIsOpen } = useNavToggle();
 
-export async function getStaticProps(): Promise<{ props: ReferencesProps }> {
-  return {
-    props: {
-      references: nxMenuApi.getReferenceApiMenuSection(),
+  const vm: { menu: Menu; packages: IntrinsicPackageMetadata[] } = {
+    menu: {
+      sections: sortCorePackagesFirst<MenuSection>(
+        getPackagesSections(menu),
+        'id'
+      ),
     },
+    packages: useMemo(
```

Comment:
Seems odd to nest `useMemo` here. Can we separate menu and packages? It also looks like menu is dependent on the menu object from props?

```ts
const menu = useMemo(() => ({
    sections: ...
}), [props.menu]);
const packages = useMemo(() = ...);
```

Then the two objects have different dependencies (`props.menu`, `props.packages`).